### ISSUE TITLE
JDK 14 support

### ISSUE THREAD

**Issue by shepherdless** [lavalink-devs/Lavalink#260]
## Description
It doesn't seem to support JDK 14

## Version info
Lavalink Build #986

Output of `java -jar Lavalink.jar --version`:
```
2020-03-24 20:37:04.794  INFO 7088 --- [           main] lavalink.server.Launcher                 :

[32m       .  [31m _                  _ _       _    [32m__ _ _
[32m      /\\ [31m| | __ ___   ____ _| (_)_ __ | | __[32m\ \ \ \
[32m     ( ( )[31m| |/ _` \ \ / / _` | | | '_ \| |/ /[32m \ \ \ \
[32m      \\/ [31m| | (_| |\ V / (_| | | | | | |   < [32m  ) ) ) )
[32m       '  [31m|_|\__,_| \_/ \__,_|_|_|_| |_|_|\_\[32m / / / /
[0m    =========================================[32m/_/_/_/[0m

        Version:        e270ec688cbad900d9de292a9e615f15c48a9a82-SNAPSHOT
        Build:          986
        Build time:     23.03.2020 20:03:20 UTC
        Branch          dev
        Commit:         e270ec6
        Commit time:    23.03.2020 19:56:15 UTC
        JVM:            14
        Lavaplayer      1.3.38

2020-03-24 20:37:04.961  INFO 7088 --- [           main] lavalink.server.Launcher                 : Starting Launcher on DESKTOP-QEERMHK with PID 7088 (C:\Users\user\Documents\Projects\Discord\kiki-bot\lavalink\Lavalink.jar started by user in C:\Users\user\Documents\Projects\Discord\kiki-bot\lavalink)
2020-03-24 20:37:04.963  INFO 7088 --- [           main] lavalink.server.Launcher                 : No active profile set, falling back to default profiles: default
2020-03-24 20:37:05.302  WARN 7088 --- [kground-preinit] o.s.h.c.j.Jackson2ObjectMapperBuilder    : For Jackson Kotlin classes support please add "com.fasterxml.jackson.module:jackson-module-kotlin" to the classpath
2020-03-24 20:37:08.061  WARN 7088 --- [           main] io.undertow.websockets.jsr               : UT026010: Buffer pool was not set on WebSocketDeploymentInfo, the default pool will be used
2020-03-24 20:37:08.094  INFO 7088 --- [           main] io.undertow.servlet                      : Initializing Spring embedded WebApplicationContext
2020-03-24 20:37:08.095  INFO 7088 --- [           main] o.s.web.context.ContextLoader            : Root WebApplicationContext: initialization completed in 2971 ms
2020-03-24 20:37:08.872  WARN 7088 --- [           main] l.server.config.SentryConfiguration      : Turning off sentry
2020-03-24 20:37:08.898  INFO 7088 --- [           main] l.s.c.AudioSendFactoryConfiguration      : OS: Windows 10, Arch: amd64
2020-03-24 20:37:08.901  INFO 7088 --- [           main] l.s.c.AudioSendFactoryConfiguration      : JDA-NAS supported system detected. Enabled native audio sending.
2020-03-24 20:37:08.903  INFO 7088 --- [           main] l.s.c.AudioSendFactoryConfiguration      : Setting buffer to 400ms
2020-03-24 20:37:08.904  INFO 7088 --- [           main] l.s.c.AudioSendFactoryConfiguration      : Setting playlist load limit to 6
2020-03-24 20:37:09.560  INFO 7088 --- [           main] c.s.d.l.tools.GarbageCollectionMonitor   : GC monitoring enabled, reporting results every 2 minutes.
2020-03-24 20:37:10.467  INFO 7088 --- [           main] o.s.s.concurrent.ThreadPoolTaskExecutor  : Initializing ExecutorService 'applicationTaskExecutor'
2020-03-24 20:37:11.195  INFO 7088 --- [           main] org.xnio                                 : XNIO version 3.3.8.Final
2020-03-24 20:37:11.229  INFO 7088 --- [           main] org.xnio.nio                             : XNIO NIO Implementation Version 3.3.8.Final
2020-03-24 20:37:11.353  INFO 7088 --- [           main] o.s.b.w.e.u.UndertowServletWebServer     : Undertow started on port(s) 2333 (http) with context path ''
2020-03-24 20:37:11.359  INFO 7088 --- [           main] lavalink.server.Launcher                 : Started Launcher in 7.546 seconds (JVM running for 8.549)
2020-03-24 20:37:21.512  INFO 7088 --- [  XNIO-1 task-1] io.undertow.servlet                      : Initializing Spring DispatcherServlet 'dispatcherServlet'
2020-03-24 20:37:21.513  INFO 7088 --- [  XNIO-1 task-1] o.s.web.servlet.DispatcherServlet        : Initializing Servlet 'dispatcherServlet'
2020-03-24 20:37:21.526  INFO 7088 --- [  XNIO-1 task-1] o.s.web.servlet.DispatcherServlet        : Completed initialization in 12 ms
2020-03-24 20:37:21.577  INFO 7088 --- [  XNIO-1 task-1] l.server.io.HandshakeInterceptorImpl     : Incoming connection from /127.0.0.1:54170
2020-03-24 20:37:21.837  INFO 7088 --- [  XNIO-1 task-1] lavalink.server.io.SocketServer          : Connection successfully established from /127.0.0.1:54170
2020-03-24 20:37:52.622  INFO 7088 --- [  XNIO-1 task-2] l.server.player.AudioLoaderRestHandler   : GET /loadtracks
2020-03-24 20:37:52.766  INFO 7088 --- [   XNIO-1 I/O-2] lavalink.server.io.SocketServer          : {"sessionId":"412fb26effd03a74186bef5e720fa682","op":"voiceUpdate","guildId":"624133384201043979","event":{"token":"e5a1dca8a5d8b30b","guild_id":"624133384201043979","endpoint":"us-east253.discord.media:80"}}
2020-03-24 20:37:53.029  INFO 7088 --- [     parallel-2] com.jcabi.manifests.Manifests            : 73 attributes loaded from 84 stream(s) in 206ms, 73 saved, 1240 ignored: ["Ant-Version", "Archiver-Version", "Automatic-Module-Name", "Bnd-LastModified", "Build-Jdk", "Build-Jdk-Spec", "Build-Time", "Built-By", "Bundle-ActivationPolicy", "Bundle-Activator", "Bundle-Category", "Bundle-Description", "Bundle-DocURL", "Bundle-License", "Bundle-ManifestVersion", "Bundle-Name", "Bundle-NativeCode", "Bundle-RequiredExecutionEnvironment", "Bundle-SymbolicName", "Bundle-Vendor", "Bundle-Version", "Copyright", "Created-By", "Export-Package", "Extension-Name", "Git-Revision", "Implementation-Build", "Implementation-Build-Date", "Implementation-Title", "Implementation-URL", "Implementation-Vendor", "Implementation-Vendor-Id", "Implementation-Version", "Import-Package", "Include-Resource", "JCabi-Build", "JCabi-Date", "JCabi-Version", "Jar-Name", "Jar-Version", "Java-Vendor", "Java-Version", "Kotlin-Runtime-Component", "Kotlin-Version", "Log4jReleaseKey", "Log4jReleaseManager", "Log4jReleaseVersion", "Magma-Version", "Main-Class", "Manifest-Version", "Multi-Release", "Originally-Created-By", "Os-Arch", "Os-Name", "Os-Version", "Private-Package", "Provide-Capability", "Require-Bundle", "Require-Capability", "Scm-Connection", "Scm-Revision", "Scm-Url", "Specification-Title", "Specification-Vendor", "Specification-Version", "Spring-Boot-Classes", "Spring-Boot-Lib", "Spring-Boot-Version", "Start-Class", "Tool", "X-Compile-Source-JDK", "X-Compile-Target-JDK", "url"]
2020-03-24 20:37:54.507  INFO 7088 --- [ader-2-thread-1] lavalink.server.player.AudioLoader       : Loaded playlist Search results for: you make me feel like dancing
2020-03-24 20:37:57.122  INFO 7088 --- [   XNIO-1 I/O-2] lavalink.server.io.SocketServer          : {"op":"play","guildId":"624133384201043979","track":"QAAAkwIAMExlbyBTYXllciAtIFlvdSBtYWtlIG1lIGZlZWwgbGlrZSBkYW5jaW5nICgxOTc2KQAJcmVzY3VlNTY3AAAAAAADW2AAC0hoU2p3VThnRXNJAAEAK2h0dHBzOi8vd3d3LnlvdXR1YmUuY29tL3dhdGNoP3Y9SGhTandVOGdFc0kAB3lvdXR1YmUAAAAAAAAAAA==","volume":100}
2020-03-24 20:38:08.512 ERROR 7088 --- [   XNIO-2 I/O-1] s.n.m.impl.connections.AudioWebSocket    : Exception in websocket connection, closing. Channel state is Websocket Channel is null

java.lang.IllegalStateException: Failed to connect to wss://us-east253.discord.media/?v=4
        at space.npstr.magma.impl.connections.hax.ClosingUndertowWebSocketClient$1.handleFailed(ClosingUndertowWebSocketClient.java:79) ~[impl-0.12.5.jar!/:na]
        at org.xnio.IoFuture$HandlingNotifier.notify(IoFuture.java:215) ~[xnio-api-3.3.8.Final.jar!/:3.3.8.Final]
        at org.xnio.AbstractIoFuture$1.run(AbstractIoFuture.java:211) ~[xnio-api-3.3.8.Final.jar!/:3.3.8.Final]
        at org.xnio.IoUtils$2.execute(IoUtils.java:70) ~[xnio-api-3.3.8.Final.jar!/:3.3.8.Final]
        at org.xnio.AbstractIoFuture.runNotifier(AbstractIoFuture.java:354) ~[xnio-api-3.3.8.Final.jar!/:3.3.8.Final]
        at org.xnio.AbstractIoFuture.runAllNotifiers(AbstractIoFuture.java:233) ~[xnio-api-3.3.8.Final.jar!/:3.3.8.Final]
        at org.xnio.AbstractIoFuture.setException(AbstractIoFuture.java:251) ~[xnio-api-3.3.8.Final.jar!/:3.3.8.Final]
        at org.xnio.FutureResult.setException(FutureResult.java:89) ~[xnio-api-3.3.8.Final.jar!/:3.3.8.Final]
        at io.undertow.websockets.client.WebSocketClient$ConnectionBuilder$2.notify(WebSocketClient.java:342) ~[undertow-core-2.0.26.Final.jar!/:2.0.26.Final]
        at org.xnio.AbstractIoFuture$1.run(AbstractIoFuture.java:211) ~[xnio-api-3.3.8.Final.jar!/:3.3.8.Final]
        at org.xnio.IoUtils$2.execute(IoUtils.java:70) ~[xnio-api-3.3.8.Final.jar!/:3.3.8.Final]
        at org.xnio.AbstractIoFuture.runNotifier(AbstractIoFuture.java:354) ~[xnio-api-3.3.8.Final.jar!/:3.3.8.Final]
        at org.xnio.AbstractIoFuture.runAllNotifiers(AbstractIoFuture.java:233) ~[xnio-api-3.3.8.Final.jar!/:3.3.8.Final]
        at org.xnio.AbstractIoFuture.setException(AbstractIoFuture.java:251) ~[xnio-api-3.3.8.Final.jar!/:3.3.8.Final]
        at org.xnio.FutureResult.setException(FutureResult.java:89) ~[xnio-api-3.3.8.Final.jar!/:3.3.8.Final]
        at org.xnio.http.HttpUpgrade$HttpUpgradeState$UpgradeResultListener.handleEvent(HttpUpgrade.java:425) ~[xnio-api-3.3.8.Final.jar!/:3.3.8.Final]
        at org.xnio.http.HttpUpgrade$HttpUpgradeState.flushUpgradeChannel(HttpUpgrade.java:369) ~[xnio-api-3.3.8.Final.jar!/:3.3.8.Final]
        at org.xnio.http.HttpUpgrade$HttpUpgradeState.access$900(HttpUpgrade.java:165) ~[xnio-api-3.3.8.Final.jar!/:3.3.8.Final]
        at org.xnio.http.HttpUpgrade$HttpUpgradeState$StringWriteListener.handleEvent(HttpUpgrade.java:396) ~[xnio-api-3.3.8.Final.jar!/:3.3.8.Final]
        at org.xnio.http.HttpUpgrade$HttpUpgradeState$StringWriteListener.handleEvent(HttpUpgrade.java:372) ~[xnio-api-3.3.8.Final.jar!/:3.3.8.Final]
        at org.xnio.ChannelListeners.invokeChannelListener(ChannelListeners.java:92) ~[xnio-api-3.3.8.Final.jar!/:3.3.8.Final]
        at org.xnio.conduits.WriteReadyHandler$ChannelListenerHandler.writeReady(WriteReadyHandler.java:65) ~[xnio-api-3.3.8.Final.jar!/:3.3.8.Final]
        at io.undertow.protocols.ssl.SslConduit$SslWriteReadyHandler.writeReady(SslConduit.java:1273) ~[undertow-core-2.0.26.Final.jar!/:2.0.26.Final]
        at io.undertow.protocols.ssl.SslConduit$4.run(SslConduit.java:430) ~[undertow-core-2.0.26.Final.jar!/:2.0.26.Final]
        at org.xnio.nio.WorkerThread.safeRun(WorkerThread.java:582) ~[xnio-nio-3.3.8.Final.jar!/:3.3.8.Final]
        at org.xnio.nio.WorkerThread.run(WorkerThread.java:466) ~[xnio-nio-3.3.8.Final.jar!/:3.3.8.Final]
Caused by: java.io.EOFException: XNIO000812: Connection closed unexpectedly
        at org.xnio.http.HttpUpgrade$HttpUpgradeState$UpgradeResultListener.handleEvent(HttpUpgrade.java:416) ~[xnio-api-3.3.8.Final.jar!/:3.3.8.Final]
        ... 10 common frames omitted






```

[spring.log](https://github.com/Frederikam/Lavalink/files/4378417/spring.log)

**Comment by freyacodes** [comment]
Relates to #258

**Comment by shepherdless** [comment]
I have applied the fixes for issue #258. Thank you. I hope JDK 14 issues are solved within a commendable time frame.
